src/Microsoft.Health.Fhir.Azure/ExportDestinationClient/AzureConnectionStringClientInitializer.cs

```diff
@@ -0,0 +1,49 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export.ExportDestinationClient;
+
+namespace Microsoft.Health.Fhir.Azure.ExportDestinationClient
+{
+    public class AzureConnectionStringClientInitializer : IExportClientInitializer<CloudBlobClient>
+    {
+        private readonly ExportJobConfiguration _exportJobConfiguration;
+        private readonly ILogger<AzureConnectionStringClientInitializer> _logger;
+
+        public AzureConnectionStringClientInitializer(IOptions<ExportJobConfiguration> exportJobConfiguration, ILogger<AzureConnectionStringClientInitializer> logger)
+        {
+            EnsureArg.IsNotNull(exportJobConfiguration?.Value, nameof(exportJobConfiguration));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _exportJobConfiguration = exportJobConfiguration.Value;
+            _logger = logger;
+        }
+
+        public Task<CloudBlobClient> GetAuthorizedClientAsync(CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(_exportJobConfiguration.StorageAccountConnection))
+            {
+                throw new ExportClientInitializerException(Resources.InvalidConnectionSettings, HttpStatusCode.BadRequest);
+            }
+
+            if (!CloudStorageAccount.TryParse(_exportJobConfiguration.StorageAccountConnection, out CloudStorageAccount cloudAccount))
+            {
+                throw new ExportClientInitializerException(Resources.InvalidConnectionSettings, HttpStatusCode.BadRequest);
+            }
+
+            CloudBlobClient blobClient = cloudAccount.CreateCloudBlobClient();
```

Comment:
if this method fails to parse the connection string, would it throw exception? if so, does it need to be translated to ExportClientInitializerException?

Reply:
The method description does not mention anything about throwing an exception. And I assumed it should be okay to call this since we successfully parse the connection string in the previous step. To be safe, I have added a try-catch block around this call also.